bitnami/kibana/templates/deployment.yaml

```diff
@@ -110,14 +147,19 @@ spec:
                   key: kibana-keystore-password
             {{- end }}
             {{- if .Values.elasticsearch.security.auth.enabled }}
-            - name: KIBANA_USERNAME
-              value: {{ .Values.elasticsearch.security.auth.kibanaUsername | quote }}
             - name: KIBANA_PASSWORD
               valueFrom:
                 secretKeyRef:
                   name: {{ include "kibana.elasticsearch.auth.secretName" . }}
                   key: kibana-password
             {{- end }}
+            {{- if and .Values.elasticsearch.security.auth.enabled .Values.elasticsearch.security.auth.createSystemUser }}
```

Comment:
I think I'm missing some context here. If I understood correctly, by default we have a `kibana_system` user without a password. If we enable ES's `security.auth` said user will use the password present obtained from `security.auth.kibanaPassword` or the secret at `security.auth.existingSecret`. 

What happens when this conditional is `true`? Does ES create a new `kibana_system` that uses `security.auth.elasticsearchPasswordSecret` as a password? I'm not seeing the need to have the `KIBANA_PASSWORD` env set twice with different secrets.

Reply:
Thanks for the explanation! The difference here is whether we are creating the `kibna_system` user with the same credentials used in ES, right?